src/lib/main.js

```diff
@@ -175,4 +178,23 @@ module.exports = async (argv, env) => {
   }
 };
 
-const executeAction = action => require(`../fn/${action}`)();
+
+const requiresInstance = actions => actions.find(actionName => {
+  const actionLib = require(`../fn/${actionName}`);
+  if (typeof actionLib === 'function') {
+    return true;
+  } else if (typeof actionLib.requiresInstance === 'boolean') {
```

Comment:
if it isn't a function, should we just assume it is an object and accept any truthy value for `requiresInstance`?

Reply:
No, because if you don't define it that will default to falsey, which is not what we want IMO.

We could either change the wording to be the opposite (`noInstanceNeeded`) to drop this logic block, or we could go through all of the existing actions and correct them into this format, allowing us to change the API to requiring you to declare that you need an instance before you're guaranteed one. However, I think defaulting to presuming you need an instance makes more sense unless we want to refactor _even further_ to not allowing actions to access instances without first declaring that they need it.

Reply:
I suggested "any defined truthy value" so `return actionLib.requiresInstance !== undefined && !!actionLib.requiresInstance;`

Reply:
I think that would mean we'd be supporting truthy but not falsy (because having `requiresInstance` as `undefined` wouldn't distinguish between that being an intended falsy value and it not being present at all) which seems confusing. 

I've changed it to what I should have done in the first place, where it probes the object and if the key is present it uses it.

Reply:
Nice! `actionLib.hasOwnProperty('requiresInstance')` is also a good option here.

Reply:
Aye, I forgot about that function, I think that's a clearer function usage, I'll swap it in :+1: